### ISSUE TITLE
Multi objects with alpha map error

### ISSUE THREAD

**Issue by brihgt** [mrdoob/three.js#12964]
##### Description of the problem 

When more than one object in a scene with alpha map(expecting to get tranparent effect), if those objects' meshes chris-cross, or overlap, or stacking(doesnt matter if thoes meshes physically touching or not touching), found the tranparent effect errors, e.g. like the attached file, the leaves are partially missing or diffuse map showing background color in some view angles.

Expected effect:
![image](https://user-images.githubusercontent.com/3327141/34338065-c3218a50-e99f-11e7-9af7-68ac75c39db6.png)

Correct effect:
![image](https://user-images.githubusercontent.com/3327141/34338085-081e41d4-e9a0-11e7-8f9d-9c17e8df6ba0.png)

In current three.js editor, I get the result as below.
![image](https://user-images.githubusercontent.com/3327141/34338108-485ae2ca-e9a0-11e7-889e-b775ee1b9a55.png)
![image](https://user-images.githubusercontent.com/3327141/34338111-4c4553f2-e9a0-11e7-9f21-f78937e50923.png)

##### Three.js version

- [ ] Dev
- [x] r89
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by WestLangley** [comment]
What you are seeing is a WebGL issue, not a three.js issue. Try
```javascript
material.alphaTest = 0.5;
material.transparent = false;
```

**Comment by RemusMar** [comment]
> What you are seeing is a WebGL issue, not a three.js issue.

That's true.
Another workaround in certain situations is to use:
```javascript
material.depthWrite = false;
```

**Comment by haroldiedema** [comment]
A technique called 'Alpha to Coverage'  would also fix this issue and works particularly well with foliage. PR #12445 would need to get merged first though.

I've added the necessary modifications to my own  three.js file as shown in #12445 to see if it works as expected. The only downside I found is that when adding over 50k planes close to each other using instanced buffer geometry, you'll notice loss in fps when looking at the foliage up close.
